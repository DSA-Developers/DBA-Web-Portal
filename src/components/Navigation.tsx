import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import Logo from '../assets/logo.png';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';

interface Props {}

const StyledTitle = styled(Navbar.Text)`
  font-family: Optima, Segoe, 'Segoe UI', Candara, Calibri, Arial, sans-serif;
  font-style: normal;
  font-weight: 400;
  margin: 0 0 0 0.5em;
  font-size: 3rem;
`;

const StyledNav = styled(Nav)`
  font-size: 2rem;
`;

const StyledNavbar = styled(Navbar)`
  background: -webkit-linear-gradient(90deg, #c7daea, #f8f9fb);
`;

const StyledLink = styled(Nav.Item)`
  margin: 0 1em 0 0;
  text-transform: uppercase;
`;
const Title = styled.div`
  font-weight: 500;
`;

export const Navigation: React.FC<Props> = ({}) => {
  return (
    <StyledNavbar className="navbar border-bottom pl-5 pr-5 py-0">
      <Navbar.Brand className="align-items-center" href="/home">
        <img src={Logo} width="80" height="80" className="d-inline-block align-top" alt="DSA Logo" />
        <StyledTitle>
          <b>DSA</b> @ UF
        </StyledTitle>
      </Navbar.Brand>
      <StyledNav className="ml-auto">
        <StyledLink>
          <Nav.Link as={Link} to="/home">
            <Title>Home</Title>
          </Nav.Link>
        </StyledLink>
        <StyledLink>
          <Nav.Link as={Link} to="/about">
            <Title>About</Title>
          </Nav.Link>
        </StyledLink>
        <StyledLink>
          <Nav.Link as={Link} to="/gallery">
            <Title>Gallery</Title>
          </Nav.Link>
        </StyledLink>
        <StyledLink>
          <Nav.Link as={Link} to="/contact">
            <Title>Contact</Title>
          </Nav.Link>
        </StyledLink>
      </StyledNav>
    </StyledNavbar>
  );
};
