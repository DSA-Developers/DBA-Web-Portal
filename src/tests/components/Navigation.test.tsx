import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallow } from 'enzyme';
import Nav from 'react-bootstrap/Nav';
import Navigation from '../../components/Navigation';

describe('Navigation', () => {
  const wrapper = shallow(<Navigation />);
  const links = wrapper.find(Nav.Link);

  test('Nav bar renders 4 links', () => {
    expect(links).toHaveLength(4);
  });

  test('Nav bar renders the proper link names', () => {
    expect(links.at(0).text()).toEqual('Home');
    expect(links.at(1).text()).toEqual('About');
    expect(links.at(2).text()).toEqual('Gallery');
    expect(links.at(3).text()).toEqual('Contact');
  });

  // snapshot test
  test('Navigation bar renders', () => {
    const props = {};
    const component = renderer.create(
      <Router>
        <Navigation {...props} />
      </Router>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
