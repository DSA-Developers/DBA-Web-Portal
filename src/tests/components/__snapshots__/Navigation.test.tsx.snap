// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar renders 1`] = `
<nav
  className="justify-content-end navbar navbar-expand navbar-light"
>
  <div
    className="navbar-nav"
    onKeyDown={[Function]}
  >
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/home"
        onClick={[Function]}
      >
        Home
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/gallery"
        onClick={[Function]}
      >
        Gallery
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
    </div>
  </div>
</nav>
`;
